PR: Add mode display

bgallois opened the pull request:
I replaced the static label "Vim:" by the currently selected mode (normal, visual, v-line or insert) enhancing clarity for the user.

ccordoba12 commented:
@bgallois, could you post a screenshot of things look now?

goanpeca commented:
Awesome work @bgallois could you include a screenshot :-) ?

bgallois commented:
It's looking like that:

![spyder_vim](https://user-images.githubusercontent.com/23381534/82360347-867ddc80-9a09-11ea-97d6-a182a67a0ef2.gif)

ccordoba12 commented:
Pretty cool @bgallois!

@goanpeca, please suggest @bgallois some colors that work for both the dark and light modes (I think the current ones don't work very well in both).

goanpeca commented:
@bgallois you have a couple of choices, you could use the colors that the codeeditor.py already defines for some things
```
        self._facecolors = {
            'warning': QColor(editor.warning_color),
            'error': QColor(editor.error_color),
            'todo': QColor(editor.todo_color),
            'breakpoint': QColor(editor.breakpoint_color),
            'occurrence': QColor(editor.occurrence_color),
            'found_results': QColor(editor.found_results_color)
            }
```

Editor in this code is `codeeditor.py``

The other option is to use the colors from the currently selected theme for the code editor, this is stored in the appearance section of the config

```python
from spyder.config.gui import get_color_scheme, is_dark_interface
theme_dictionary = get_color_scheme(CONF.get('appearance', 'selected')
```

Also using is dark interface could help you select which theme so we could probably stick to `spyder` and `spyder/dark`themes depending on `is_dark_interface`.

Let me know how it goes.

----

Also a little nitpick, could we make the width of the label fixed and equal to the largest word?

Having the width change on mode change could get annoying pretty fast :-)

Thanks for the hard work!

ccordoba12 commented:
> Also a little nitpick, could we make the width of the label fixed and equal to the largest word?

Totally agree with this.

bgallois commented:
I fixed the label width and changed the colors based on the dark or light spyder theme.

![spyder_vim](https://user-images.githubusercontent.com/23381534/82896168-c5ee8080-9f55-11ea-953f-ab46e9bacaff.gif)

goanpeca commented:
Awesome! Thanks for the fix!

ccordoba12 commented:
@bgallois, please solve the merge conflicts so we can merge this one.

goanpeca commented:
Thanks for working on this!
